File: Coding in Python/03rd March 2026/Find Kth Bit in Nth Binary String.py
#Find Kth Bit in Nth Binary String
class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        s = []
        s.append("0")
        for i in range(1, n):
            temp = ""
            for ch in s[i - 1]:
                if ch == '0':
                    temp += '1'
                else:
                    temp += '0'
            s.append(s[i - 1] + "1" + temp[::-1])
        return s[n - 1][k - 1]

File: coding/03rd March 2026/Find Kth Bit in Nth Binary String.java
//Find Kth Bit in Nth Binary String
class Solution 
{
    public char findKthBit(int n, int k) 
    {
        StringBuilder[] s = new StringBuilder[n];
        s[0] = new StringBuilder("0");
        for(int i = 1; i < n; i++)
        {
            StringBuilder temp = new StringBuilder();
            for(int j = 0; j < s[i - 1].length(); j++)
            {
                if(s[i - 1].charAt(j) == '0')
                    temp.append('1');
                else
                    temp.append('0');
            }
            s[i] = new StringBuilder();
            s[i].append(s[i - 1]);
            s[i].append("1");
            s[i].append(temp.reverse());
        }
        return s[n - 1].charAt(k - 1);
    }
}
